crates/dojo-core/src/world/world_contract.cairo

```diff
@@ -1274,5 +1414,77 @@ pub mod world {
                 layout
             )
         }
+
+        fn write_model_ty(ref self: ContractState, model_selector: felt252, ty: @Ty) {
+            let mut serialized = ArrayTrait::new();
+            Serde::serialize(ty, ref serialized);
+
+            // TODO: clean the Vec first
+
+            let mut model_ty = self.models.entry(model_selector).ty;
+            loop {
+                match serialized.pop_front() {
+                    Option::Some(e) => {
+                        model_ty.append().write(e);
+                    },
+                    Option::None => { break; }
+                };
+            };
+        }
+
+        fn write_model_layout(
+            ref self: ContractState, model_selector: felt252, layout: @Layout
+        ) {
+            let mut serialized = ArrayTrait::new();
+            Serde::serialize(layout, ref serialized);
+
+            // TODO: clean the Vec first
+
+            let mut model_layout = self.models.entry(model_selector).layout;
+            loop {
+                match serialized.pop_front() {
+                    Option::Some(e) => {
+                        model_layout.append().write(e);
+                    },
+                    Option::None => { break; }
+                };
+            };
+        }
+
+        fn read_model_ty(self: @ContractState, model_selector: felt252) -> Ty {
+            let mut serialized = ArrayTrait::new();
+
+            let model_ty = self.models.entry(model_selector).ty;
+
+            // TODO: Cairo 2.8 : to replace by "for i in 0..model_ty.len()"
+            let mut i = 0;
+            loop {
+                if i >= model_ty.len() { break; }
+                serialized.append(model_ty.at(i).read());
+
+                i += 1;
+            };
+            let mut serialized = serialized.span();
+
+            Serde::<Ty>::deserialize(ref serialized).unwrap()
+        }
+
+        fn read_model_layout(self: @ContractState, model_selector: felt252) -> Layout {
+            let mut serialized = ArrayTrait::new();
+
+            let model_layout = self.models.entry(model_selector).layout;
+
+            // TODO: Cairo 2.8 : to replace by "for i in 0..model_ty.len()"
+            let mut i = 0;
+            loop {
+                if i >= model_layout.len() { break; }
+                serialized.append(model_layout.at(i).read());
+
+                i += 1;
+            };
+            let mut serialized = serialized.span();
+
+            Serde::<Layout>::deserialize(ref serialized).unwrap()
```

Comment:
This part may be one function with generic argument to match both `Layout` and `Ty`?